### PR TITLE
Use global search query syntax highlighting classes for query input

### PR DESCRIPTION
**Pull request by fkling**
Follow up to #38331

This commit removes the custom CSS classes for syntax highlighting the
query in the search query input and instead uses the global classes
introduced in 721efd3aad8bba51eb8e9efaeb9398f419ee7b5d.

To make that easier I refactored the code and extracted the CSS class
name generation into its own function, applying a more table-driven
approach.

## Test plan

Type query in main query input and inspect query token DOM elements.

### CODE REVIEW

**Comment by fkling** [comment]
@rvantonder If you rather prefer the logic to be more explicit (individual `case` statements for token types and kinds), let me know.

**Review by rvantonder — APPROVED** [review]
> If you rather prefer the logic to be more explicit (individual case statements for token types and kinds), let me know.

No ways lol I was just doing the first obvious thing and got tired copying everything so kind of just went "eh, good enough". Thanks for the refactor.

File: client/shared/src/search/query/decoratedToken.ts [hunk 1]
```diff
@@ -1088,6 +1088,74 @@ export const decorate = (token: Token): DecoratedToken[] => {
     return decorated
 }
 
+const tokenKindToCSSName: Record<MetaRevisionKind | MetaRegexpKind | MetaPredicateKind | MetaStructuralKind, string> = {
```
Comment (rvantonder) [inline client/shared/src/search/query/decoratedToken.ts:1091]: in fact it crossed my mind translate this by converting camlCase to hyphenated lower case, but record too is good enough :-)

Comment (fkling) [reply client/shared/src/search/query/decoratedToken.ts:1091]: Exactly what I thought (including the "good enough" :D)

**Pull request merged**